### ISSUE TITLE
Feature request: Ignore delete

### ISSUE THREAD

**Issue by nicholasgcoles** [mathandpencil/django-bulk-sync#7]
Great library, thanks! Saved me a ton of time.

Wondering if there is a supported `ignore_delete` argument that could optionally be provided so that the sync does not remove entries it can't find?

Thinking it could be something like this:
```
def bulk_sync(new_models, key_fields, filters, batch_size=None, fields=None, ignore_delete=False):
    """ Combine bulk create, update, and delete.  Make the DB match a set of in-memory objects.

    `new_models`: Django ORM objects that are the desired state.  They may or may not have `id` set.
    `key_fields`: Identifying attribute name(s) to match up `new_models` items with database rows.  If a foreign key
            is being used as a key field, be sure to pass the `fieldname_id` rather than the `fieldname`.
    `filters`: Q() filters specifying the subset of the database to work in.  Use `None` or `[]` if you want to sync against the entire table.
    `batch_size`: passes through to Django `bulk_create.batch_size` and `bulk_update.batch_size`, and controls
            how many objects are created/updated per SQL query.
    `fields`: (optional) list of fields to update. If not set, will sync all fields that are editable and not auto-created.
    `ignore_delete`: (optional) boolean to ignore deleting "stale" objects

    """
    db_class = new_models[0].__class__

    if fields is None:
        # Get a list of fields that aren't PKs and aren't editable (e.g. auto_add_now) for bulk_update
        fields = [field.name
                  for field in db_class._meta.fields
                  if not field.primary_key and not field.auto_created and field.editable]

    with transaction.atomic():
        objs = db_class.objects.all()
        if filters:
            objs = objs.filter(filters)
        objs = objs.only("pk", *key_fields).select_for_update()

        def get_key(obj):
            return tuple(getattr(obj, k) for k in key_fields)

        obj_dict = {get_key(obj): obj for obj in objs}

        new_objs = []
        existing_objs = []
        for new_obj in new_models:
            old_obj = obj_dict.pop(get_key(new_obj), None)
            if old_obj is None:
                # This is a new object, so create it.
                # Make sure the primary key field is clear.
                new_obj.pk = None
                new_objs.append(new_obj)
            else:
                new_obj.id = old_obj.id
                existing_objs.append(new_obj)

        db_class.objects.bulk_create(new_objs, batch_size=batch_size)
        db_class.objects.bulk_update(existing_objs, fields=fields, batch_size=batch_size)

        # delete stale ones...
        if ignore_delete is False:
            objs.filter(pk__in=[_.pk for _ in list(obj_dict.values())]).delete()

        assert len(existing_objs) == len(new_models) - len(new_objs)

        stats = {"created": len(new_objs), "updated": len(new_models) - len(new_objs),
                 "deleted": len(obj_dict) if not ignore_delete else 0,
                 "ignored": len(obj_dict) if ignore_delete else 0}

        logger.debug(
            "{}: {} created, {} updated, {} deleted.".format(
                db_class.__name__, stats["created"], stats["updated"], stats["deleted"], stats["ignored"]
            )
        )

    return {"stats": stats}
```

**Comment by ses4j** [comment]
@nicholasgcoles `ignore_delete` seems useful and appropriate.  Maybe I prefer the name `skip_deletes`?  For completeness, any thoughts about including `skip_creates` or `skip_updates`?

Feel free to submit a PR.

**Comment by ses4j** [comment]
@nicholasgcoles I assume you saw someone submitted (or caused someone to submit) a PR for this earlier today, just FYI.  #9

**Comment by ses4j** [comment]
Released as v2.1.0.
